### PR TITLE
 bug-1408178: Sort rules history by data version

### PR DESCRIPTION
**Pull request by andela-ookwuolisa**
This PR sorts the rules' history by the data version

### CODE REVIEW

File: ui/app/js/controllers/rules_controller.js [hunk 1]
```diff
@@ -132,7 +132,10 @@ function($scope, $routeParams, $location, $timeout, Rules, Search, $modal, $rout
     // will be inserted based on the priority in the Scheduled Change.
     // Rules that have Scheduled updates or deletes will remain sorted on their current priority
     // because it's more important to make it easy to assess current state than future state.
-    if (rule.priority === null || rule.priority === undefined) {
+    if(rule.timestamp){
```
Comment (mozbhearsum) [inline ui/app/js/controllers/rules_controller.js]: Please base this decision on $scope.rule_id instead - that's what we do everywhere else.

File: ui/app/js/controllers/rules_controller.js [hunk 2]
```diff
@@ -170,12 +170,23 @@ function($scope, $routeParams, $location, $timeout, Rules, Search, $modal, $rout
...
+  };
+
   $scope.orderRules = function(rule) {
     // Rules are sorted by priority. Rules that are pending (ie: still just a Scheduled Change)
     // will be inserted based on the priority in the Scheduled Change.
     // Rules that have Scheduled updates or deletes will remain sorted on their current priority
     // because it's more important to make it easy to assess current state than future state.
-    if (rule.priority === null || rule.priority === undefined) {
+
+    if($scope.returnRuleType() === "Revision"){
```
Comment (mozbhearsum) [inline ui/app/js/controllers/rules_controller.js]: While this works, I think you're overcomplicating it. You can simply check if `$scope.rule_id` is set to know whether or not we're on the Rules History page, like we do at https://github.com/mozilla/balrog/blob/0b0e7e53c6c399aca913b49ecef53966d76dba01/ui/app/js/controllers/rules_controller.js#L60.

File: ui/app/js/controllers/rules_controller.js [hunk 3]
```diff
@@ -170,12 +170,23 @@ function($scope, $routeParams, $location, $timeout, Rules, Search, $modal, $rout
...
 
+  $scope.returnRuleType = function() {
+    var firstRuleId = $scope.rules[0].rule_id;
+    return $scope.rules.every(function(rule) {
+      return rule.rule_id === firstRuleId;
+    }) ? "Revision" : "All Rules";
+  };
+
   $scope.orderRules = function(rule) {
     // Rules are sorted by priority. Rules that are pending (ie: still just a Scheduled Change)
```
Comment (mozbhearsum) [inline ui/app/js/controllers/rules_controller.js]: Please keep this comment just above in the "else" block -- it doesn't relate to the new code you've added.

File: ui/app/js/controllers/rules_controller.js (hunk 2 above)
Comment (andela-ookwuolisa) [reply ui/app/js/controllers/rules_controller.js]: Hi, Ben, I thought about that initially, but I observed `$scope.rule_id` is always set for both Rules History and All Rules. The reason why I had to check if `$scope.rule_id` has the same value for each rule or not.

Comment (mozbhearsum) [reply ui/app/js/controllers/rules_controller.js]: Are you sure about this? $scope.rule_id comes from $routeParams.id. When we're on /rules, there's no id set as far as I know...

**Pull request merged**